digiwf-libs/digiwf-camunda-prometheus/digiwf-camunda-prometheus-core/src/main/java/de/muenchen/oss/digiwf/camunda/prometheus/FniAndEdeMetricsProvider.java

```diff
@@ -3,21 +3,36 @@
 import io.prometheus.client.CollectorRegistry;
 import io.prometheus.client.Gauge;
 import lombok.RequiredArgsConstructor;
+import org.camunda.bpm.engine.HistoryService;
 import org.camunda.bpm.engine.ManagementService;
+import org.camunda.bpm.engine.RepositoryService;
+import org.camunda.bpm.engine.repository.ProcessDefinition;
+
+import java.util.List;
 
 @RequiredArgsConstructor
 public class FniAndEdeMetricsProvider implements MetricsProvider {
 
     private final ManagementService managementService;
+    private final RepositoryService repositoryService;
+    private final HistoryService historyService;
 
     private Gauge fniCount;
     private Gauge edeCount;
 
     @Override
     public void updateMetrics() {
         // Get total number of FNIs
-        this.fniCount.labels("total").set(this.managementService.createMetricsQuery().name("activity-instance-start").sum());
-
+        this.fniCount
+            .labels("total", "total")
+            .set(this.managementService.createMetricsQuery().name("activity-instance-start").sum());
+
+        // Get FNIs by process definition (this only includes the currently deployed definitions and may be lower than the total count)
+        repositoryService.createProcessDefinitionQuery().list().forEach(processDefinition ->
```

Comment:
Wir haben ganz viele Definitionen. Kann das in Prod viel Last erzeugen?

Reply:
Die Frage ist wie häufig wir scrapen...